sql/core/src/main/scala/org/apache/spark/sql/DataFrameNaFunctions.scala

```diff
@@ -132,62 +132,42 @@ final class DataFrameNaFunctions private[sql](df: DataFrame) {
    *
    * @since 1.3.1
    */
-  def fill(value: Double): DataFrame = fill(value, df.columns)
-
-  /**
-   * Returns a new [[DataFrame]] that replaces null values in string columns with `value`.
-   *
-   * @since 1.3.1
-   */
-  def fill(value: String): DataFrame = fill(value, df.columns)
```

Comment:
You won't be able to remove public API methods like this. Why is this necessary if the idea is to handle Long differently?